run-until-success-with-backoff

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -euo pipefail
+
+set -x
+
+i=0
+max_retries=20
+until "$@"; do
+    ((i+=1))
+    if [[ $i == "$max_retries" ]]; then
```

Comment:
Can we start printing messages as soon as something takes longer than 10 seconds, to give some additional visibility to things that are slow.

Reply:
Done.